FTBFS dist-git in Fedora Rawhide

The test-suite fails:

```Creating /tmp/tmppbkk88hn/new.txt
b'127.0.0.1 - - [18/Jun/2024 08:17:22] "POST /cgi-bin/upload.cgi HTTP/1.1" 200 -\nTraceback (most recent call last):\n  File "/tmp/tmppbkk88hn/cgi-bin/upload.cgi", line 9, in <module>\n    import cgi\nModuleNotFoundError: No module named \'cgi\'\n127.0.0.1 - - [18/Jun/2024 08:17:22] CGI script exit code 1\n'
_______________ UploadTest.test_accepts_sha_512_hash_1_rpms_pkg ________________

a = (<test_upload_script.UploadTest testMethod=test_accepts_sha_512_hash_1_rpms_pkg>,)
kw = {}

    @wraps(func)
    def standalone_func(*a, **kw):
>       return func(*(a + p.args), **p.kwargs, **kw)

/usr/lib/python3.13/site-packages/parameterized/parameterized.py:620: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_upload_script.py:233: in test_accepts_sha_512_hash
    self.assertEqual(resp, 'File new.txt size 7 SHA512 %s stored OK\n' % NEW_SHA512)
E   AssertionError: '' != 'File new.txt size 7 SHA512 d3d67bc3e38489[122 chars]OK\n'
E   + File new.txt size 7 SHA512 d3d67bc3e3848925892de9b132c9ff4054a05c9dbc7b4366d16b5c6b87c898df60da162e9ec415d4dc16470128ef52c11c44fc06da2841543ddeb351b10e9fb2 stored OK
```

Python has been updated to 3.13 in Rawhide, might be related.

--- a/dist-git/dist-git.spec
+++ b/dist-git/dist-git.spec
@@ -60,8 +60,14 @@ BuildRequires:  python3-requests
 
 # this should be Requires but see https://bugzilla.redhat.com/show_bug.cgi?id=1833810
 Recommends: moreutils
-
 %endif
+
+%if 0%{?fedora} && 0%{?fedora} >= 41
+# The `cgi` module was removed from the Python 3.13 standard library
+BuildRequires:  python3-legacy-cgi
+Requires:       python3-legacy-cgi
+%endif
+
 
 %description
 DistGit is a Git repository specifically designed to hold RPM

